src/printer.js

```diff
@@ -330,23 +330,26 @@ function printExpression(path, options, print) {
           path.call(print, "offset")
         ]);
       case "offsetlookup": {
-        const isOffsetNumberNode = node.offset && node.offset.kind === "number";
+        const hasOffset = node.offset;
```

Comment:
If it's called `hasOffset`, it should be `Boolean(node.offset)` instead. Maybe you could just drop the extra variable and use node.offset directly.